components/chat_widget/src/components/ocs-chat/ocs-chat.tsx

```diff
@@ -309,6 +323,31 @@ export class OcsChat {
     this.parsedStarterQuestions = this.parseJSONProp(this.starterQuestions, 'starter questions');
   }
 
+  private async initializeTranslations() {
+    let customTranslationsObj: Partial<TranslationStrings> | undefined;
+
+    if (this.translationsUrl) {
+        customTranslationsObj = await this.loadTranslationsFromUrl(this.translationsUrl);
+    }
+
+    this.currentLanguage = this.language || getBrowserLanguage();
+    this.translationManager = new TranslationManager(this.currentLanguage, customTranslationsObj);
+  }
+
+  private async loadTranslationsFromUrl(url: string): Promise<Partial<TranslationStrings>> {
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+      const translations = await response.json();
+      return translations as Partial<TranslationStrings>;
+    } catch (error) {
+      console.error('Error loading translations from URL:', error);
+      throw error;
```

Comment:
I think this will prevent the component from loading altogether. It would be better to fallback to the default translations.